src/test/test-time-util.c

```diff
@@ -238,6 +238,11 @@ TEST(format_timespan) {
         test_format_timespan_accuracy(1);
         test_format_timespan_accuracy(USEC_PER_MSEC);
         test_format_timespan_accuracy(USEC_PER_SEC);
+
+        /* See issue #23928. */
+        _cleanup_free_ char *buf;
+        assert_se(buf = new(char, 5));
+        format_timespan(buf, 5, 100005, 1000);
```

Comment:
I suspect this might need to be prefixed with `(void) ` to make compilers happy, if the results should indeed be ignored.

Reply:
Interesting. I have no idea why my compiler does not complain about that...